components/button/styles.ts

```diff
@@ -1,105 +1,627 @@
+      },
+      success: {
+        bg: palette.cNeutral8,
+        border: palette.success,
+        color: palette.success,
+      },
+      warning: {
+        bg: palette.cNeutral8,
+        border: palette.warning,
+        color: palette.warning,
+      },
+      error: {
+        bg: palette.cNeutral8,
+        border: palette.error,
+        color: palette.error,
+      },
+    },
+    text: {
+      primary: {
+        bg: addColorAlpha(palette.cTheme5, 0.12),
+        border: 'transparent',
+        color: palette.cTheme7,
+      },
+      default: {
+        bg: palette.cNeutral2,
+        border: 'transparent',
+        color: palette.cNeutral6,
+      },
+      secondary: {
+        bg: addColorAlpha(palette.cTheme3, 0.24),
+        border: 'transparent',
+        color: palette.cTheme4,
+      },
+      success: {
+        bg: addColorAlpha(palette.success, 0.24),
+        border: 'transparent',
+        color: palette.success,
+      },
+      warning: {
+        bg: addColorAlpha(palette.warning, 0.24),
+        border: 'transparent',
+        color: palette.warning,
+      },
+      error: {
+        bg: addColorAlpha(palette.error, 0.24),
+        border: 'transparent',
+        color: palette.error,
+      },
+    },
+  }
+  const colors: { [key in ButtonTypes]?: ButtonColorGroup } = category[variant]
+  const defaultColor = colors.default as ButtonColorGroup
+  return colors[color] || defaultColor
+}
+
+export const getButtonDisabledColors = (
+  palette: ZeitUIThemesPalette,
+  props: ButtonProps,
+): ButtonColorGroup => {
+  const { variant } = props
+  const category = {
+    solid: {
+      bg: palette.cNeutral3,
+      border: palette.cNeutral3,
+      color: palette.cNeutral5,
+    },
+    lined: {
+      bg: palette.cNeutral8,
+      border: palette.cNeutral3,
+      color: palette.cNeutral5,
+    },
+    text: {
+      bg: 'transparent',
+      border: 'transparent',
+      color: palette.cNeutral5,
+    },
+  }
+  return category[variant]
+}
+
+export const getButtonGhostColors = (
+  palette: ZeitUIThemesPalette,
+  color: ButtonTypes,
+): ButtonColorGroup | null => {
+  const colors: { [key in ButtonTypes]?: ButtonColorGroup } = {
     secondary: {
-      bg: palette.foreground,
+      bg: palette.background,
       border: palette.foreground,
-      color: palette.background,
+      color: palette.foreground,
     },
     success: {
-      bg: palette.success,
+      bg: palette.background,
       border: palette.success,
-      color: '#fff',
+      color: palette.success,
     },
     warning: {
-      bg: palette.warning,
+      bg: palette.background,
       border: palette.warning,
-      color: '#fff',
+      color: palette.warning,
     },
     error: {
-      bg: palette.error,
+      bg: palette.background,
       border: palette.error,
-      color: '#fff',
-    },
-    abort: {
-      bg: 'transparent',
-      border: 'transparent',
-      color: palette.accents_5,
+      color: palette.error,
```

Comment:
We can use color provided by our designer, it's recommended to use cNetural7 and cNetural8 as background and foreground color.

If there's style in this component that not in the design prototype, better sync with the designer.

Reply:
OK